packages/reputation-miner/ReputationMiner.js

```diff
@@ -1313,67 +1346,59 @@ class ReputationMiner {
-    await db.run("CREATE TABLE IF NOT EXISTS colonies ( address text NOT NULL UNIQUE )");
-    await db.run("CREATE TABLE IF NOT EXISTS skills ( skill_id INTEGER PRIMARY KEY )");
-    await db.run(
+    // Not regularly used, so not preparing them and saving the statement for reuse
+    await this.db.prepare("CREATE TABLE IF NOT EXISTS users ( address text NOT NULL UNIQUE )").run();
+    await this.db.prepare("CREATE TABLE IF NOT EXISTS reputation_states ( root_hash text NOT NULL UNIQUE, n_leaves INTEGER NOT NULL)").run();
+    await this.db.prepare("CREATE TABLE IF NOT EXISTS colonies ( address text NOT NULL UNIQUE )").run();
+    await this.db.prepare("CREATE TABLE IF NOT EXISTS skills ( skill_id INTEGER PRIMARY KEY )").run();
+    await this.db.prepare(
       `CREATE TABLE IF NOT EXISTS reputations (
         reputation_rowid text NOT NULL,
         colony_rowid INTEGER NOT NULL,
         skill_id INTEGER NOT NULL,
         user_rowid INTEGER NOT NULL,
         value text NOT NULL
       )`
-    );
+    ).run();
 
     // Do we have to do a database upgrade, from when we renamed n_nodes to n_leaves?
-    const nNodesColumn = await db.all("SELECT * FROM PRAGMA_TABLE_INFO('reputation_states') WHERE name='n_nodes';")
+    const nNodesColumn = await this.db.prepare("SELECT * FROM PRAGMA_TABLE_INFO('reputation_states') WHERE name='n_nodes';").all()
     if (nNodesColumn.length === 1) {
-      await db.run("ALTER TABLE 'reputation_states' RENAME COLUMN n_nodes to n_leaves");
-      const check1 = await db.all("SELECT * FROM PRAGMA_TABLE_INFO('reputation_states') where name='n_nodes'")
-      const check2 = await db.all("SELECT * FROM PRAGMA_TABLE_INFO('reputation_states') where name='n_leaves'")
+      await this.db.prepare("ALTER TABLE 'reputation_states' RENAME COLUMN n_nodes to n_leaves").run();
+      const check1 = await this.db.prepare("SELECT * FROM PRAGMA_TABLE_INFO('reputation_states') where name='n_nodes'").all()
+      const check2 = await this.db.prepare("SELECT * FROM PRAGMA_TABLE_INFO('reputation_states') where name='n_leaves'").all()
       if (check1.length !== 0 || check2.length !== 1){
         console.log('Unexpected result of DB upgrade');
         process.exit();
       }
       console.log('n_nodes -> n_leaves database upgrade complete');
     }
-    await db.close();
   }
 
   async resetDB() {
-    const db = await sqlite.open({filename: this.dbPath, driver: sqlite3.Database});
-    await db.run(`DROP TABLE IF EXISTS users`);
-    await db.run(`DROP TABLE IF EXISTS colonies`);
-    await db.run(`DROP TABLE IF EXISTS skills`);
-    await db.run(`DROP TABLE IF EXISTS reputations`);
-    await db.run(`DROP TABLE IF EXISTS reputation_states`);
-    await db.close();
+    // Again, not regularly used, so not preparing and saving the statements for reuse.
+    await this.db.prepare(`DROP TABLE IF EXISTS users`).run();
```

Comment:
How is this preparing and saving if we don't save the statements to some constant?

Reply:
Perhaps a misreading of the comment, which explicitly says we are _not_ preparing and saving the statement?

Reply:
Yup, correct.